crates/bevy_reflect/derive/src/where_clause_options.rs

```diff
@@ -157,19 +178,57 @@ impl<'a, 'b> WhereClauseOptions<'a, 'b> {
             let bevy_reflect_path = self.meta.bevy_reflect_path();
             let reflect_bound = self.reflect_bound();
 
-            // `TypePath` is always required for active fields since they are used to
-            // construct `NamedField` and `UnnamedField` instances for the `Typed` impl.
-            // Likewise, `GetTypeRegistration` is always required for active fields since
-            // they are used to register the type's dependencies.
-            Some(self.active_fields.iter().map(move |ty| {
-                quote!(
-                    #ty : #reflect_bound
-                        + #bevy_reflect_path::TypePath
-                        // Needed for `Typed` impls
-                        + #bevy_reflect_path::MaybeTyped
-                        // Needed for `GetTypeRegistration` impls
-                        + #bevy_reflect_path::__macro_exports::RegisterForReflection
-                )
+            // Get the identifiers of all type parameters.
+            let type_param_idents = self
+                .meta
+                .type_path()
+                .generics()
+                .type_params()
+                .map(|type_param| type_param.ident.clone())
+                .collect::<Vec<Ident>>();
+
+            // Do any of the identifiers in `idents` appear in `token_stream`?
+            fn is_any_ident_in_token_stream(idents: &[Ident], token_stream: TokenStream) -> bool {
+                for token_tree in token_stream {
```

Comment:
I'm pretty sure this is the real reason we just slapped the bounds on all field types. In most cases these streams should be super small so I really doubt it should be a huge performance hit, if any (the other changes probably offset any possible hit anyways haha). Do you happen to have any benchmarks on how this affects compile times for fields with larger generic types?

Reply:
To make sure I understand the question: you're worried about the cost of iterating through the tokens of the field types? I haven't measured that, I seriously doubt the cost would be anything more than negligible, and the cost savings you get from the rest of the compiler not processing the unnecessary bounds would be much larger.

I suspect the reason that the bounds if on all fields is that it was easier that way, because proc macros are a bit of a pain to write in general.